javascript/modules/client/src/internal/client-core.ts

```diff
@@ -244,4 +253,46 @@ export abstract class ClientCore implements IClientCore {
         }
       });
   }
+
+  public pin(input: string | Uint8Array): Promise<string> {
+    if (!this.ipfs)
+      return Promise.reject(new Error("IPFS client is not initialized"));
+    return this.ipfs
+      .add(input)
+      .then(res => res.path)
+      .catch(e => {
+        throw new Error("Could not pin data: " + (e?.message || ""));
+      });
+  }
+
+  public async fetchBytes(cid: string) {
+    if (!this.ipfs) throw new Error("IPFS client is not initialized");
+    try {
+      let chunks: Uint8Array[] = [];
+      let totalArrayLength = 0;
+      for await (const chunk of this.ipfs.cat(cid)) {
+        chunks.push(chunk);
+        totalArrayLength += chunk.length;
+      }
+      var mergedArray = new Uint8Array(totalArrayLength);
+      let lastIndex = 0;
+      chunks.forEach(chunk => {
+        mergedArray.set(chunk, lastIndex);
+        lastIndex += chunk.length;
+      });
```

Comment:
It's a bit more efficient to avoid creating new function contexts for every loop iteration

```suggestion
      for (const chunk of chunks) {
        mergedArray.set(chunk, lastIndex);
        lastIndex += chunk.length;
      }
```